### PR TITLE
Fix IndexError for hdf5 file import for single instance analysis files

### PR DESCRIPTION
**Pull request by shrivaths16**
### Description
While performing inference for single instances, there seems to be no track names generated. This is causing `IndexError` while trying to load a HDF5 file since, the number of tracks is 1 and the `track_names` is empty. So we add a condition to check this and create a single track accordingly.

### Types of changes

- [x] Bugfix
- [ ] New feature
- [ ] Refactor / Code style update (no logical changes)
- [ ] Build / CI changes
- [ ] Documentation Update
- [ ] Other (explain)

### Does this address any currently open issues?
[list open issues here]

### Outside contributors checklist

- [ ] Review the [guidelines for contributing](https://github.com/talmolab/sleap/blob/develop/docs/CONTRIBUTING.md) to this repository
- [ ] Read and sign the [CLA](https://github.com/talmolab/sleap/blob/develop/sleap-cla.pdf) and add yourself to the [authors list](https://github.com/talmolab/sleap/blob/develop/AUTHORS)
- [ ] Make sure you are making a pull request against the **develop** branch (not *main*). Also you should start *your branch* off *develop*
- [ ] Add tests that prove your fix is effective or that your feature works
- [ ] Add necessary documentation (if appropriate)

#### Thank you for contributing to SLEAP!
:heart:

## Summary by CodeRabbit

- **New Features**
    - Enhanced data import by importing `FileHandle` for improved functionality.
    - Introduced new fixtures `small_robot_3_frame_hdf5`, `single_fly_hdf5`, and `single_fly_mp4_vid` for testing.
    - Included constants `TEST_SMALL_ROBOT3_FRAME_H5`, `TEST_SINGLE_FLY_H5`, and `TEST_SINGLE_FLY` for test scenarios.
    - Added a new test function `test_sleap_analysis_read` to read Sleap analysis files using `SleapAnalysisAdaptor`.

### CODE REVIEW

**Review by talmo — CHANGES_REQUESTED** [review]
Add test

File: tests/io/test_formats.py [hunk 1]
```diff
@@ -19,6 +19,19 @@
 from sleap.gui.app import MainWindow
 from sleap.gui.state import GuiState
 from sleap.info.write_tracking_h5 import get_nodes_as_np_strings
+from sleap.io.format.sleap_analysis import SleapAnalysisAdaptor
+
+
+def test_sleap_analysis_read(single_fly_mp4_vid, single_fly_hdf5):
```
Comment (roomrys) [inline tests/io/test_formats.py:25]: I know we needed a single-instance hdf5 file, but do you mind reusing one of the videos we already have? I don't think we really need to add a whole new video to out test data.

File: tests/data/videos/single_fly.mp4
Comment (roomrys) [inline tests/data/videos/single_fly.mp4:1]: Remove

File: tests/data/hdf5_format_v1/single_fly.h5
Comment (roomrys) [inline tests/data/hdf5_format_v1/single_fly.h5:1]: Maybe you can make a really small h5 file using our 3 frame small_robot.mp4?

**Review by roomrys — CHANGES_REQUESTED** [review]
Thanks for adding the test. For the fixture, I think it's a bit big. Maybe we can use one of the videos we already have (proposing small_robot.mp4) and also making a really small analysis file from that.

File: tests/io/test_formats.py (hunk 1 above)
Comment (shrivaths16) [reply tests/io/test_formats.py]: Yes, I have used a smaller one in my latest commit

**Review by talmo — APPROVED** [review]

**Pull request merged**